### ISSUE TITLE
`MOUSEINPUT` is of the wrong type

### ISSUE THREAD

**Issue by pentamassiv** [microsoft/windows-rs#2908]
### Summary

It is the same issue as https://github.com/microsoft/windows-rs/issues/2708. Even after updating the metadata in https://github.com/microsoft/windows-rs/commit/b62b802bae534fdaed3fa25b6838dc3001b6d084, the issue still persists.

In version 0.51, `windows::Win32::UI::Input::KeyboardAndMouse::MOUSEINPUT.mouseData` was of type `i32`. Starting in version 0.52, it is of type `u32`. According to the documentation, the type is DWORD. However it also states:

> If dwFlags contains MOUSEEVENTF_WHEEL, then mouseData specifies the amount of wheel movement. A positive value indicates that the wheel was rotated forward, away from the user; a negative value indicates that the wheel was rotated backward, toward the user. One wheel click is defined as WHEEL_DELTA, which is 120.

 ([source](https://learn.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-mouseinput)).

**Comment by kennykerr** [comment]
`MOUSEINPUT` is originally defined as follows:

```C++
typedef struct tagMOUSEINPUT {
    LONG    dx;
    LONG    dy;
    DWORD   mouseData;
    DWORD   dwFlags;
    DWORD   time;
    ULONG_PTR dwExtraInfo;
} MOUSEINPUT, *PMOUSEINPUT, FAR* LPMOUSEINPUT;
```

The metadata defines it as follows:

```C#
public struct MOUSEINPUT
{
	public int dx;
	public int dy;
	public uint mouseData;
	public MOUSE_EVENT_FLAGS dwFlags;
	public uint time;
	public UIntPtr dwExtraInfo;
}
```

So `mouseData` looks correct to me.

**Comment by pentamassiv** [comment]
It looks like there is a lot of going back and forth between the two types (https://github.com/microsoft/win32metadata/issues?q=is%3Aissue+mouseinput+is%3Aclosed).  It was changed to i32 because you have to be able to use negative values to scroll up (https://github.com/microsoft/win32metadata/issues/933) and a special case was added to the metadata. Later that seems to have been reverted.

**Comment by kennykerr** [comment]
Yes, it seems that way. I haven't been tracking that, but I believe the value is correct as of now. I'll transfer to the Win32 metadata repo for their input.
